generate_td_to_bq_mapping.py

```diff
@@ -147,21 +152,25 @@ def map_table_references(client: bigquery.Client, table_references: List[str],
         " must add it before you can continue"
 
     table_map = {}
-    for tr in table_references:
-        split_tr = tr.split('.')
+    for reference in table_references:
+        table_id = reference.get('table', reference.get('view'))
+        split_tr = table_id.split('.')
         split_tr_length = len(split_tr)
         assert split_tr_length == 3 or split_tr_length == 2, "Malformed Table" \
-                                                             f" Reference. {tr} has {split_tr_length} elements. " \
-                                                             f"It should have either 2 or 3."
+                                                                 f" Reference. {reference} has {split_tr_length} elements. " \
+                                                                 f"It should have either 2 or 3."
 
         if len(split_tr) == 3:
             table = split_tr[2]
         else:
             table = split_tr[1]
 
-        logger.info(f"  Creating mapping: '{tr}' -> " \
+        logger.info(f"  Creating mapping: '{reference}' -> " \
                     f"'{mapped_dataset}.{table}'")
-        table_map[tr] = f"{mapped_dataset}.{table}"
+        table_type = reference.keys()
+        if table_type == 'view':
+            mapped_dataset = f'{mapped_dataset}_views'
```

Comment:
This should be read from the `business_unit_dataset_map.csv`. Not hard-coded. We need to update the `business_unit_map` to return
```
{"CREDIT": 
  { 
   "view": "pp_credit_views",
  "table": "pp_credit"
  }
} 
```

Then, the `dataset_map = business_unit_map.get(business_unit)` call will return a dictionary like:
```
  { 
   "view": "pp_credit_views",
  "table": "pp_credit"
  }
```

We can then set `mapped_dataset = dataset_map[table_type]`